### ISSUE TITLE
training curve inconsistent with EasyR1

### ISSUE THREAD

**Issue by Elliott011** [volcengine/verl#1967]
## Question
I use verl to train Qwen2.5-VL-7B-Instruct with GRPO on Geometry3K. However, the training curve is significantly inconsistent with the training curve using the [EasyR1](https://github.com/hiyouga/EasyR1) repository under the same configuration.

## Setting & results of verl

The code version is https://github.com/volcengine/verl/commit/54b2677f72f6720f80480d72af7f47eee74b4dc.

The command is `bash examples/grpo_trainer/run_qwen2_5_vl-7b.sh`. The script is the same with that provided in the repo, except for some settings such as experiment name.

```
set -x
ENGINE=${1:-vllm}
# If you are using vllm<=0.6.3, you might need to set the following environment variable to avoid bugs:
# export VLLM_ATTENTION_BACKEND=XFORMERS


export WANDB_API_KEY=xxxxxxxxxxx
EXPERIMENT_NAME=qwen2_5_vl_7b_geo3k_grpo
TIMESTAMP=$(date +"%Y-%m-%d_%H-%M-%S")
LOG_DIR=log/${EXPERIMENT_NAME}/${TIMESTAMP}
mkdir -p ${LOG_DIR}


python3 -m verl.trainer.main_ppo \
    algorithm.adv_estimator=grpo \
    data.train_files=training_data/geo3k/train.parquet \
    data.val_files=training_data/geo3k/test.parquet \
    data.train_batch_size=512 \
    data.max_prompt_length=1024 \
    data.max_response_length=2048 \
    data.filter_overlong_prompts=True \
    data.truncation='error' \
    data.image_key=images \
    actor_rollout_ref.model.path=Qwen/Qwen2.5-VL-7B-Instruct \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.actor.ppo_mini_batch_size=128 \
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=10 \
    actor_rollout_ref.actor.use_kl_loss=True \
    actor_rollout_ref.actor.kl_loss_coef=0.01 \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \
    actor_rollout_ref.actor.entropy_coeff=0 \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.actor.fsdp_config.param_offload=False \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=False \
    actor_rollout_ref.actor.checkpoint.contents=['model','optimizer','extra','hf_model'] \
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=20 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=2 \
    actor_rollout_ref.rollout.name=$ENGINE \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.6 \
    actor_rollout_ref.rollout.enable_chunked_prefill=False \
    actor_rollout_ref.rollout.enforce_eager=False \
    actor_rollout_ref.rollout.free_cache_engine=False \
    actor_rollout_ref.rollout.n=5 \
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=20 \
    actor_rollout_ref.ref.fsdp_config.param_offload=True \
    algorithm.use_kl_in_reward=False \
    trainer.critic_warmup=0 \
    trainer.logger=['console','wandb'] \
    trainer.project_name='verl' \
    trainer.experiment_name=$EXPERIMENT_NAME \
    trainer.n_gpus_per_node=8 \
    trainer.nnodes=1 \
    trainer.save_freq=10 \
    trainer.test_freq=5 \
    trainer.total_epochs=120 \
    trainer.default_local_dir=checkpoints/verl/$EXPERIMENT_NAME/$TIMESTAMP \
    ray_init.num_cpus=32 \
    2>&1 | tee -a ${LOG_DIR}/all.log
```

The training curve is:
![Image](https://github.com/user-attachments/assets/3076597b-c5be-4bdd-8439-3cc77e588da7)

The accuracy of the training set suddenly decreases after rising to around 0.6.

## Setting & results of [EasyR1](https://github.com/hiyouga/EasyR1)

However, if I use the [EasyR1](https://github.com/hiyouga/EasyR1) repo (version: https://github.com/hiyouga/EasyR1/commit/daa75cb00ad1ab241ca41664387e9168788c7e55) to run the same training, the accuracy of the training set increases steadily, as shown in the following figure.

![Image](https://github.com/user-attachments/assets/7ab13023-fb51-4baf-8357-2775b9ff5134)

The training scripts is 

```
set -x

export WANDB_API_KEY=xxxxxxxxxxx
EXPERIMENT_NAME=qwen2_5_vl_7b_geo_grpo
TIMESTAMP=$(date +"%Y-%m-%d_%H-%M-%S")
LOG_DIR=log/${EXPERIMENT_NAME}/${TIMESTAMP}

MODEL_PATH=Qwen/Qwen2.5-VL-7B-Instruct  # replace it with your local file path

FORMAT_PROMPT="""You FIRST think about the reasoning process as an internal monologue and then provide the final answer.
 The reasoning process MUST BE enclosed within <think> </think> tags. The final answer MUST BE put in \boxed{}."""

mkdir -p ${LOG_DIR}
python3 -m verl.trainer.main \
    config=examples/config.yaml \
    data.train_files=hiyouga/geometry3k@train \
    data.val_files=hiyouga/geometry3k@test \
    data.format_prompt="${FORMAT_PROMPT}" \
    worker.actor.model.model_path=${MODEL_PATH} \
    worker.rollout.tensor_parallel_size=2 \
    worker.rollout.gpu_memory_utilization=0.85 \
    trainer.total_episodes=120 \
    trainer.experiment_name=$EXPERIMENT_NAME \
    trainer.n_gpus_per_node=8 \
    trainer.save_checkpoint_path=checkpoints/${EXPERIMENT_NAME}/${TIMESTAMP} \
    2>&1 | tee -a ${LOG_DIR}/all.log
```

**Comment by xytian1008** [comment]
How about on the validation set?

**Comment by zheng980629** [comment]
Have you identified and fixed the bug? I'm encountering the same issue.

**Comment by Elliott011** [comment]
> How about on the validation set?

Performance on validation set, trained by `verl` (version https://github.com/volcengine/verl/commit/54b2677f72f6720f80480d72af7f47eee74b4dc.):
<img width="1707" height="447" alt="Image" src="https://github.com/user-attachments/assets/9a4bde31-0fd6-470f-9420-cea8b98cd844" />

Performance on validation set, trained by `EasyR1` (version: https://github.com/hiyouga/EasyR1/commit/daa75cb00ad1ab241ca41664387e9168788c7e55):
<img width="3792" height="1992" alt="Image" src="https://github.com/user-attachments/assets/b9e47d89-812d-4750-8771-d865040b25e4" />

**Comment by Elliott011** [comment]
> Have you identified and fixed the bug? I'm encountering the same issue.

Not yet.

**Comment by Kirrito-k423** [comment]
### 中文
根据本次定时任务对 comments 和 timeline 的检查，这个 issue 命中默认超时关闭规则：最近一次非当前运行账号（Kirrito-k423）的公开回复或公开引用是 2025-07-26T09:24:17Z，已经超过 6 个月。证据：latest_non_self_reply_or_reference_at=2025-07-26T09:24:17Z; kind=comment; actor=Elliott011; reference=https://github.com/verl-project/verl/issues/1967#issuecomment-3121559145。检查来源：https://api.github.com/repos/verl-project/verl/issues/1967/comments 和 https://api.github.com/repos/verl-project/verl/issues/1967/timeline。因此按定时清理标准先关闭；如果作者或社区成员仍能提供新的复现、证据或需求背景，可以继续评论请求重开或创建新 issue。

### English
In this scheduled pass, I checked the issue comments and timeline. This issue matches the default stale-issue closure rule: the latest public reply or public reference from an account other than the current runner (Kirrito-k423) was at 2025-07-26T09:24:17Z, which is more than 6 months ago. Evidence: latest_non_self_reply_or_reference_at=2025-07-26T09:24:17Z; kind=comment; actor=Elliott011; reference=https://github.com/verl-project/verl/issues/1967#issuecomment-3121559145. Checked sources: https://api.github.com/repos/verl-project/verl/issues/1967/comments and https://api.github.com/repos/verl-project/verl/issues/1967/timeline. I am closing it under the scheduled cleanup rule. If the author or community can provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
